### ISSUE TITLE
How can I get the current httpContext user in a handler?

### ISSUE THREAD

**Issue by VictorioBerra** [soxtoby/SlackNet#143]
Tried adding the following to my program:

```csharp
builder.Services.AddHttpContextAccessor();
// Add useful interface for accessing the ActionContext outside a controller.
builder.Services.AddSingleton<IActionContextAccessor, ActionContextAccessor>();
```

## handler:

```csharp
 public class LinkSlackAccount : IEventHandler<MessageEvent>
{
        public async Task Handle(MessageEvent slackEvent)
        {
             this.actionContextAccessor.ActionContext // <----- NULL!!!!
        }
}
```

**Comment by soxtoby** [comment]
Thanks for the question, @VictorioBerra.

Slack requests aren't handled by MVC, so there is no controller or action to get the context for. There probably won't be a HTTP context either, since the request is responded to _before_ running any handlers, and in socket mode ASP.NET is bypassed entirely.

If you can describe what you were going to use the action context for, I may be able to suggest an alternative.

**Comment by VictorioBerra** [comment]
@soxtoby Ah duh I was not thinking when I posted this. Its an event from slack of course their wont be a user.

So the way my app works currently:

1. User logs into web app through Azure B2C (AAD) and gets a cookie.
2. User clicks a button that generates a Guid we store in the DB along with their B2C Object ID (user ID)
3. User pastes the guid into slack.
4. EventHandler fires, uses the guid to look up the user, and then adds some metadata about the slack user to their account
5. Reply to slack with "User Bob has linked their slack account"

This process lets us link a user of our app with their slack account.

What I ended up doing was on step 4 was pulling their B2C Object ID from the Guid table, calling Azure with the was the MS Graph API to get the user details like Display name and using that in the reply to slack. IE: `$"Web user {graphUser.DisplayName} has linked their slack account (${(await _slack.Users.Info(slackEvent.User)).Name})"`.

So all is well.
